### DISCUSSION TITLE
third party signature api call to sign a document process

### DISCUSSION THREAD

**Discussion by yatinkumarverma** [AOT-Technologies/forms-flow-ai#1573] [Q&A]
Hello Community Experts,

I'm a new user to the formsflow application and building a process wherein the initiator will attach a pdf document for the assignee to read the document in the formsflow application and sign with the signature image of 200kb and then send this document to the further approvers to sign the document with the docusign signatures or with their predefined images stored in database/files to sign the document. Kindly let me know how to achieve it.

With Regards

**Comment by devmangel** [comment]
Hello @yatinkumarverma 

It sounds like you're on the right track to integrating a digital signature process into your formsflow application. Here are the steps you might consider to achieve your goal:

- API Integration: First, ensure that the third-party API (like DocuSign) is accessible from your formsflow environment. You'll need the API keys and proper authorization set up to interact with the DocuSign API.

- Attach PDF: When an initiator uploads a PDF document, store it temporarily within your application or a designated storage service.
 
- Generate Signature Request: Use the DocuSign API to create a signature request. This request will involve specifying the PDF document, the location on the document where signatures are required, and the signees (whose predefined signatures are stored).
 
- Sign Document: Signees will either use a predefined signature image or use DocuSign's interface to sign the document. Ensure that each signature image does not exceed the size limit you've mentioned (200kb).
 
- Document Management: After all required signatures are collected, the API should return the signed document. You can then forward this document to the next approver or finalize the process by storing the signed document in your database or another secure location.
 
- Notification and Tracking: Implement notifications to inform signees when their signature is required and track the status of document approvals through the formsflow application.
 
Feel free to ask further if you need more detailed guidance or specific issues arise!
